aptos-move/move-examples/shared_account/sources/shared_account.move

```diff
@@ -0,0 +1,136 @@
+// This module demonstrates a basic shared account that could be used for NFT royalties
+// Users can (1) create a shared account (2) disperse the coins to multiple creators
+module shared_account::SharedAccount {
+    use std::errors;
+    use std::signer;
+    use std::vector;
+    use AptosFramework::Account;
+    use AptosFramework::Coin;
+
+    // struct Share records the address of the share_holder and the numerator and the denominator of a fraction
+    struct Share has store {
+        share_holder: address,
+        numerator: u64,
+    }
+
+    // Resource representing a shared account
+    struct SharedAccount has key {
+        share_record: vector<Share>,
```

Comment:
Why not use a map of share holder address to the number of shares (numerator) here?

Reply:
oh yea it's because we cannot iterate a `SimpleMap` but we need to iterate the address <> number of shares mapping in the disperse() function

Reply:
How about IterableTable?

Reply:
oh yea we thought about that but it seems like it's more costly than `SimpleMap` because it has copy, store & drop?

Reply:
I see. I guess it's fine for now to use a vector then 👍